python/pyspark/sql/utils.py

```diff
@@ -490,3 +492,49 @@ def remote_only(func: Union[Callable, property]) -> Union[Callable, property]:
     else:
         func._remote_only = True  # type: ignore[attr-defined]
         return func
+
+
+T = TypeVar("T")
+
+
+class CachedAccessor(Generic[T]):
```

Comment:
There will be a followup to consolidate CachedAccessors between the Pandas API on Spark and the one here